### PR TITLE
Add deprecation notices to functions that will go away in v0.3.0.

### PR DESCRIPTION
**Pull request by iphydf**
This change is [Reviewable](https://reviewable.io/reviews/toktok/c-toxcore/798)

### CODE REVIEW

**Comment by zoff99** [comment]
what if you miss any events? why are all these going to be removed?

**Comment by iphydf** [comment]
How do you miss events?

**Comment by zoff99** [comment]
a bug,
or just dont want event but rather "ask" later about some thing

**Comment by iphydf** [comment]
If toxcore fails to emit the callback, then most likely the value is also not stored inside the tox instance, so the getter doesn't give you anything more. The positive thing is that toxcore no longer needs to keep track of this information that is most likely stored in the client anyway. I'd like to make toxcore itself more "core"-ish, smaller interface, and provide additional functionality through extra libraries. Separation of concerns and all that :).

**Comment by robinlinden** [comment]
:lgtm_strong:

---

Reviewed 2 of 2 files at r1.
Review status: all files reviewed at latest revision, all discussions resolved, some commit checks failed.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/toktok/c-toxcore/798#-:-L5jtO3A-lvAWEGc4dEV:b-jbsf8z)*

**Comment by iphydf** [comment]
Added client maintainers to see if they can accept this change.

**Comment by zoff99** [comment]
LGTM.
yet i don't agree with the change to remove api call for clients to get some infos. having only the events is not good. more complex clients code.

**Comment by iphydf** [comment]
I agree, it makes the client code of very simple clients a bit more complex. I don't think it'll do much for already complex clients, because they will have found no use in getting e.g. the byte representation of the status message from scratch each time. They will have stored it in their own data structure in a form they can handle nicely (e.g. as UTF-16 string). I'm vaguely planning to pull out a bit of functionality from toxcore and put it in a client helper library for C clients (probably not very useful for other languages). That could store things like that and also be allowed file I/O, which toxcore isn't allowed.

**Comment by sudden6** [comment]
:lgtm_strong:

---

Reviewed 2 of 2 files at r2.
Review status: all files reviewed at latest revision, all discussions resolved, some commit checks failed.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/toktok/c-toxcore/798#-:-L5vE7GaqFTWsO9djk_a:b-jbsf8z)*

File: toxcore/tox.api.h [hunk 1]
```diff
@@ -443,7 +455,9 @@ static class options {
    * @deprecated The memory layout of this struct (size, alignment, and field
    * order) is not part of the ABI. To remain compatible, prefer to use $new to
    * allocate the object and accessor functions to set the members. The struct
-   * will become opaque (i.e. the definition will become private) in v0.2.0.
+   * will become opaque (i.e. the definition will become private) in v0.2.x.
```
Comment (nurupo) [inline toxcore/tox.api.h]: API/ABI breaking change in `v0.2.x`? I thought that our versioning was that of Semantic Versioning but with the leading zero removed. Making such change in a `v0.2.x` violates that.

**Comment by zoff99** [comment]
:lgtm_strong:

---

Review status: 0 of 2 files reviewed at latest revision, 1 unresolved discussion.

---



*Comments from [Reviewable](https://beta.reviewable.io/reviews/toktok/c-toxcore/798#-:-L5y0L2m2E78fwpn-aMa:b-jbsf8z)*

**Comment by zoff99** [comment]
Reviewed 2 of 2 files at r3.
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---



*Comments from [Reviewable](https://beta.reviewable.io/reviews/toktok/c-toxcore/798)*

**Comment by JFreegman** [comment]
:lgtm_strong:

---

Reviewed 2 of 2 files at r4.
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/toktok/c-toxcore/798#-:-L6-51youambNtYrbaXr:b-jbsf8z)*

**Pull request merged**